mettle/src/mettle.c

```diff
@@ -197,6 +199,38 @@ struct mettle *mettle(void)
 	return NULL;
 }
 
+const char * mettle_get_machine_id(void)
+{
+  struct utsname utsbuf;
+  struct dirent *data;
+  static char machine_id[MAX_PATH] = "";
```

Comment:
If this is static, then we should use an early-out:
``` c
if (strlen(machine_id) > 0) { return machine_id; }
```
Or something similar.

If we don't want to use an early-out, then this can't be static as it'll result in subsequent calls appending to each thanks to the use of `strncat`.